netlist/src/design.rs

```diff
@@ -505,6 +503,18 @@ impl Design {
             Name(name.into(), value.into());
     }
 
+    pub fn add_adc(&self, a: impl Into<Value>, b: impl Into<Value>, ci: impl Into<Net>) -> Value {
+        let a = a.into();
+        let b = b.into();
+        let ci = ci.into();
+        assert_eq!(a.len(), b.len());
```

Comment:
Part of the reason why we wouldn't want to do it this way is because this duplicates a part of `validate()`.